spec/lib/jll_property_valuation/api_spec.rb

```diff
@@ -1,951 +1,823 @@
 require "spec_helper"
 
 RSpec.describe JllPropertyValuation::Api do
-  let(:api_key) { TODO: <your token here> }
+  let(:api_key) { TODO: Your API Key here }
   let(:subject) {
     JllPropertyValuation::Api.new(
-      username: TODO: <your username here>,
-      password: TODO: <your password here>,
-      api_key: api_key
+      TODO: Username here,
+      TODO: Password here,
+      api_key
     )
   }
 
+  describe '#auth_ping' do
+    it "should return a hash with correct keys" do
+      expect(subject.auth_ping().keys).to include("Status", "Message", "Data")
+    end
+  end
+
+  describe "#auth_login" do
+    it "returns the correct keys" do
+      expect(subject.auth_login().keys).to include("Status", "Message", "Data")
+    end
+
+    it "returns the correct status" do
+      expect(subject.auth_login()["Status"]).to eq("ok")
+    end
+
+    it "returns the correct message" do
+      expect(subject.auth_login()["Message"]).to eq(nil)
+    end
+
+    it "returns the correct data" do
```

Comment:
this is pointless.

Lets assume the keys in "data" are "foo" and "bar" some day, what does this help ?

Don't test their system, test ours. Test that a successful login returns a data object, because presumably we need that. What's is in the Data object is not something you can control or change, so there is no point in testing that.
If you want to make sure you get back a token, then write a spec that checks that an ok login includes a token.